### PR TITLE
Day 10 - Add find_next_neighbors

### PR DESCRIPTION
**Pull request by Rosein**

### CODE REVIEW

File: day_10/task_1/pipe_maze.cpp [hunk 1]
```diff
@@ -1,4 +1,7 @@
 #include "pipe_maze.hpp"
+#include <cassert>
+#include <string>
+#include <vector>
```
Comment (dariannax3) [inline day_10/task_1/pipe_maze.cpp:4]: vector is included in hpp file

File: day_10/task_1/pipe_maze.cpp [hunk 2]
```diff
@@ -14,3 +17,87 @@ MazePoint find_starting_point(const PipeMaze& pipe_maze)
...
+
+bool is_connected_to_north(const char candi_south_pipe)
+{
+    static const std::string pipes_to_north{"|LJ"};
+    return pipes_to_north.contains(candi_south_pipe);
+}
+
+bool is_connected_to_south(const char candi_north_pipe)
+{
+    static const std::string pipes_to_south{"|F7"};
```
Comment (dariannax3) [inline day_10/task_1/pipe_maze.cpp:29]: Q: Is 'pipes to south ' always looks the same?

Comment (Rosein) [reply day_10/task_1/pipe_maze.cpp:29]: A: Yes, it is always the same by design of the rules of the task. We have limited set of all kind of pipes and `pipes_to_south` is a subset of pipe kinds, which has at least one end pointed to south.

File: day_10/task_1/pipe_maze.cpp (hunk 1 above)
Comment (Rosein) [reply day_10/task_1/pipe_maze.cpp:4]: True

Comment (dariannax3) [reply day_10/task_1/pipe_maze.cpp:4]: comment

**Pull request merged**